File: A7/ToyLanguage/Model/ConstExp.cs
using System;
namespace ToyLanguage.Model
{
    public class ConstExp : Exp
    {
        private int number;

        public ConstExp(int value)
        {
            this.number = value;
        }
        public override int eval(IDict<String, int> dict)
        {
            return this.number;
        }
        public override string ToString()
        {
            return this.number.ToString();
        }
    }
}

File: A8-ToyLanguageInterpretatorGUI/src/Model/ConstExp.java
package Model;

public class ConstExp extends Exp {
    private int number;

    public ConstExp(int value) {
        this.number = value;
    }

    int eval(MyIDictionary<String, Integer> tbl, MyIRandIntKeyDict<Integer> heap) {
        return number;
    }

    @Override
    public String toString() {
        return String.valueOf(number);
    }
}